pydjamodb/connection.py

```diff
@@ -0,0 +1,122 @@
+import time
+
+from django.conf import settings
+
+from pynamodb.connection.table import TableConnection as BaseTableConnection
+from pynamodb.constants import TABLE_STATUS, ACTIVE
+from pynamodb.exceptions import TableDoesNotExist, TableError
+
+from botocore.client import ClientError
+
+
+class TableConnection(BaseTableConnection):
+
+    def __init__(self, table_name,
+                 region=settings.PYDJAMODB_DATABASE.get('AWS_REGION'),
+                 host=settings.PYDJAMODB_DATABASE.get('HOST'),
+                 connect_timeout_seconds=None,
+                 read_timeout_seconds=None,
+                 max_retry_attempts=None,
+                 base_backoff_ms=None,
+                 max_pool_connections=None,
+                 extra_headers=None,
+                 aws_access_key_id=settings.PYDJAMODB_DATABASE.get('AWS_ACCESS_KEY_ID'),
+                 aws_secret_access_key=settings.PYDJAMODB_DATABASE.get('AWS_SECRET_ACCESS_KEY'),
+                 aws_session_token=settings.PYDJAMODB_DATABASE.get('AWS_SESSION_TOKEN')):
+        table_name = '{}-{}'.format(settings.PYDJAMODB_DATABASE['TABLE_PREFIX'], table_name)
+        super().__init__(
+            table_name,
+            region=region,
+            host=host,
+            connect_timeout_seconds=connect_timeout_seconds,
+            read_timeout_seconds=read_timeout_seconds,
+            max_retry_attempts=max_retry_attempts,
+            base_backoff_ms=base_backoff_ms,
+            max_pool_connections=max_pool_connections,
+            extra_headers=extra_headers,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token
+        )
+
+    def create_table(self,
```

Comment:
@trebidav @radimsuckr  point in time recovery

Reply:
Wouldn't it be possible for you to use boto3 waiters? https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#DynamoDB.Waiter.TableExists

Reply:
I will try it good tip